coveralls/reporter.py

```diff
@@ -12,8 +12,14 @@
 class CoverallReporter:
     """Custom coverage.py reporter for coveralls.io."""
 
-    def __init__(self, cov, conf):
+    def __init__(self, cov, conf, base_dir=None):
         self.coverage = []
+        if base_dir:
+            self.base_dir = base_dir.replace(os.path.sep, '/')
+            if self.base_dir[-1] != '/':
+                self.base_dir += '/'
+        else:
+            self.base_dir = None
```

Comment:
nit:
```suggestion
    def __init__(self, cov, conf, base_dir=''):
        self.coverage = []
        self.base_dir = base_dir.replace(os.path.sep, '/')
        if self.base_dir[-1] != '/':
            self.base_dir += '/'
```

Keeping `base_dir` unconditionally as a string is a bit easier to reason about (fewer cases to enumerate) and will allow us to simplify the handling both here and in `parse_file()`, which no longer needs to have as many conditional checks (see comment below).

Reply:
You need the `if base_dir` test regardless or the index operation will fail, so I thought it better to just do the minimal amount of work (the assignment) in the common case...  no need to do the replace or the indexing operations 99% of the time
```
>>> base_dir = ''
>>> if base_dir[-1] != '/':
...     base_dir += '/'
... 
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
IndexError: string index out of range
```

Reply:
Ah, good call, makes sense to leave as you have it (though the above example could be replaced with `if not base_dir.endswith('/')` to avoid that IndexError, FYI!).

Reply:
ya that's a good point.. I guess I thought checking that single last char was more direct than using a string func, but I see the advantage

thanks for the thorough review